code/datums/components/connect_containers.dm

```diff
@@ -0,0 +1,62 @@
+/// This component behaves similar to connect_loc_behalf, but it's nested and hooks a signal onto all MOVABLES containing this atom.
+/datum/component/connect_containers
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+
+	/// An assoc list of signal -> procpath to register to the loc this object is on.
+	var/list/connections
+	var/atom/movable/tracked
+
+/datum/component/connect_containers/Initialize(atom/movable/tracked, list/connections)
+	. = ..()
+	if (!ismovable(tracked))
+		return COMPONENT_INCOMPATIBLE
+
+	src.connections = connections
+	src.tracked = tracked
+
+/datum/component/connect_containers/InheritComponent(datum/component/component, original, atom/movable/tracked, list/connections)
+	// Not equivalent. Checks if they are not the same list via shallow comparison.
+	if(!compare_list(src.connections, connections))
+		return
```

Comment:
Is denying component change if the args are different actually behavior you want?

Reply:
Yes because it can lead to issues because of the main limitation of datum component signal system, which is that only one specific proc can be associated to a registered signal at a time. Like all `connect_` components and elements it's the parent (the datum the component is attached to) that actually registers the connections signals, otherwise this component could have easily been of the dupe allowed category.
I should add some stack_trace call here though...